src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.cs

```diff
@@ -275,108 +275,167 @@ public static void ValidateHeaderNameCharacters(string headerCharacters)
             }
         }
 
-        public static unsafe ConnectionOptions ParseConnection(StringValues connection)
+        public static ConnectionOptions ParseConnection(StringValues connection)
         {
+            // Keep-alive
+            const ulong lowerCaseKeep = 0x0000_0020_0020_0020; // Don't lowercase hyphen
+            const ulong keepAliveStart = 0x002d_0070_0065_0065; // 4 chars "eep-"
+            const ulong keepAliveMiddle = 0x0076_0069_006c_0061; // 4 chars "aliv"
+            const ushort keppAliveEnd = 0x0065; // 1 char "e"
```

Comment:
I was about to merge but I just saw this. Nit:

```suggestion
            const ushort keepAliveEnd = 0x0065; // 1 char "e"
```